sample_project/Assets/StarterAssets/ThirdPersonController/Scripts/ThirdPersonController.cs

```diff
@@ -115,20 +115,23 @@ public class ThirdPersonController : MonoBehaviour
         private bool _hasAnimator;
 
         private RaycastHit hitBelow;
+        private Vector2 lastMousePosition;
 
-        private bool IsCurrentDeviceMouse
+
+        public bool IsKeyboard()
         {
-            get
+            var mousePosition = Mouse.current.position.ReadValue() - lastMousePosition;
+            
+            if (mousePosition == Vector2.zero)
```

Comment:
```suggestion
            return mousePosition != Vector2.zero;
```
Never do the pattern `if (bool) {return true; } else { return false; }`

Reply:
This doesn't relate to the touch controls added in this PR and can be roped into the General Samples Code Cleanup issue.
This file also shouldn't be edited in this PR so will revert.